### PR TITLE
feat(components): migrate various components to use rtl

### PR DESCRIPTION
**Pull request by jacobdolsen**
- TertiaryButton
- Select
- RegionLinks
- RegionCard
- Radio
- QuickLink

## Description

Updated various components to use RTL

💁‍

## Related issues

Continuation of https://app.shortcut.com/qantasaccommodation/story/82661

💪

## Looks like this

🤨

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [x] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), [Marcel](https://github.com/Anthropic), [Sam](https://github.com/SamuSan), [Long](https://github.com/KieraDOG)) or a Roo UI working-group human.
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [x] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

File: src/components/QuickLink/QuickLink.test.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import QuickLink from './';
+
+const text = 'links text';
+const href = '/';
+
+test('renders correctly', () => {
+  render(<QuickLink text={text} href={href} />);
+
+  expect(screen.getByText('links text')).toBeInTheDocument();
```
Comment (ajlu) [inline src/components/QuickLink/QuickLink.test.jsx:12]:
```suggestion
  expect(screen.getByRole('link', { name: 'links text' })).toBeInTheDocument();
```
By default, use `getByRole` when possible following RTL's query priority - https://testing-library.com/docs/queries/about#priority (Same for line 29)

As a result, can also then kill the `renders the Flex component as an anchor tag` test

File: src/components/RegionLinks/RegionLinks.test.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+test('shows the heading', () => {
+  render(<RegionLinks {...props} />);
+  expect(
+    screen.getByRole('heading', { level: 3, name: 'Victoria' }),
+  ).toBeInTheDocument();
+});
+
+test('shows the regions', () => {
+  render(<RegionLinks {...props} />);
+  expect(screen.getByText('Melbourne')).toHaveAttribute(
```
Comment (ajlu) [inline src/components/RegionLinks/RegionLinks.test.jsx:43]:
```suggestion
  expect(screen.getByRole('link', { name: 'Melbourne' })).toHaveAttribute(
```
Same for the tests here

File: src/components/Select/Select.test.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { render, screen, axe } from 'testUtils';
+
+import Select from './';
+import Label from '../Label';
+
+test('renders correctly', () => {
+  render(<Select>Options</Select>);
+
+  expect(screen.getByText('Options')).toBeInTheDocument();
```
Comment (ajlu) [inline src/components/Select/Select.test.jsx:10]:
```suggestion
  expect(screen.getByRole('combobox')).toBeInTheDocument();
```
Check the list of roles here - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques#roles

File: src/components/TertiaryButton/TertiaryButton.test.jsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { render, screen, axe } from 'testUtils';
+import { cleanup } from '@testing-library/react';
+
+import TertiaryButton from './TertiaryButton';
+
+afterEach(cleanup);
+test('renders correctly', () => {
+  render(<TertiaryButton>Hello world</TertiaryButton>);
+
+  expect(screen.getByText('Hello world')).toBeInTheDocument();
```
Comment (ajlu) [inline src/components/TertiaryButton/TertiaryButton.test.jsx:12]:
```suggestion
  expect(screen.getByRole('button, { name: 'Hello world' })).toBeInTheDocument();
```

File: src/components/TertiaryButton/TertiaryButton.test.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+/* eslint-env jest */
```
Comment (ajlu) [inline src/components/TertiaryButton/TertiaryButton.test.jsx:1]: What is this comment required for?

File: src/components/TertiaryButton/TertiaryButton.test.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import React from 'react';
+import { render, screen, axe } from 'testUtils';
+import { cleanup } from '@testing-library/react';
+
+import TertiaryButton from './TertiaryButton';
+
+afterEach(cleanup);
```
Comment (ajlu) [inline src/components/TertiaryButton/TertiaryButton.test.jsx:8]:
```suggestion
```
cleanup isn't required?

File: src/components/TertiaryButton/TertiaryButton.test.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+test('renders correctly', () => {
+  render(<TertiaryButton>Hello world</TertiaryButton>);
+
+  expect(screen.getByText('Hello world')).toBeInTheDocument();
+});
+
+test('has no accessibility errors', async () => {
+  let { container } = render(<TertiaryButton>Hello world</TertiaryButton>);
```
Comment (ajlu) [inline src/components/TertiaryButton/TertiaryButton.test.jsx:16]:
```suggestion
  const { container } = render(<TertiaryButton>Hello world</TertiaryButton>);
```

File: src/components/TertiaryButton/TertiaryButton.test.jsx (hunk 5 above)
Comment (jacobdolsen) [reply src/components/TertiaryButton/TertiaryButton.test.jsx:1]: Might have been generated when I renamed file to .jsx. Removing now

File: src/components/Select/Select.test.jsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+test('renders select in disabled state when disabled', () => {
+  render(<Select disabled>Options</Select>);
+
+  expect(screen.getByRole('combobox')).toBeDisabled();
+});
+
+test('renders correctly when readOnly', () => {
+  render(<Select readOnly>Options</Select>);
+
+  expect(screen.getByRole('combobox')).toBeInTheDocument();
```
Comment (ajlu) [inline src/components/Select/Select.test.jsx:55]:
```suggestion
  expect(screen.getByRole('combobox')).toHaveAttribute('readonly');
```
to assert the prop?

File: src/components/TertiaryButton/TertiaryButton.test.jsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  render(<TertiaryButton align="center">Hello world</TertiaryButton>);
+  expect(screen.getByRole('button', { name: 'Hello world' })).toHaveStyle(
+    'text-align: center',
+  );
+});
+
+test('disabled renders correctly', () => {
+  render(<TertiaryButton disabled>Hello world</TertiaryButton>);
+  expect(
+    screen.getByRole('button', { name: 'Hello world' }).closest('button'),
```
Comment (ajlu) [inline src/components/TertiaryButton/TertiaryButton.test.jsx:36]:
```suggestion
    screen.getByRole('button', { name: 'Hello world' }),
```
don't need `closest` anymore?

**Review by ajlu — APPROVED** [review]
Great work! 🚀

File: src/components/Select/Select.test.jsx (hunk 8 above)
Comment (jacobdolsen) [reply src/components/Select/Select.test.jsx:55]: Ah yup, missed this one

**Pull request merged**